tardis/__init__.py

```diff
@@ -99,51 +99,39 @@ def logging_state(log_state, tardis_config, specific):
     specific: boolean
         Allows to set specific logging levels. Logs of the `log_state` level would be output.
     """
+
     if "debug" in tardis_config:
-        if specific or tardis_config["debug"]["specific"]:
-            specific = True
-        else:
-            specific = False
-
-        if tardis_config["debug"]["log_state"] or log_state:
-            if (
-                log_state.upper() == "CRITICAL"
-                and tardis_config["debug"]["log_state"]
-            ):
-                logging_level = tardis_config["debug"]["log_state"]
-            elif log_state:
-                logging_level = log_state
-                if tardis_config["debug"]["log_state"] and log_state:
-                    print(
-                        "log_state is defined both in Functional Argument & YAML Configuration {debug section}"
-                    )
-                    print(
-                        f"log_state = {log_state} will be used for Log Level Determination\n"
-                    )
-            else:
-                logging_level = tardis_config["debug"]["log_state"]
+        specific = (
+            tardis_config["debug"]["specific"] if specific is None else specific
+        )
+
+        logging_level = (
+            log_state if log_state else tardis_config["debug"]["log_state"]
+        )
+        logging_level = logging_level.upper()
+
+        if not logging_level in LOGGING_LEVELS.keys():
+            raise ValueError(
+                f"Passed Value for log_state = {logging_level} is Invalid. Must be one of the following {list(LOGGING_LEVELS.keys())}"
+            )
     else:
-        logging_level = log_state
-        specific = specific
+        tardis_config["debug"] = {}
+        logging_level = tardis_config["debug"]["log_state"]
 
     loggers = [
         logging.getLogger(name) for name in logging.root.manager.loggerDict
     ]
-    if logging_level.upper() in LOGGING_LEVELS.keys():
+    if logging_level in LOGGING_LEVELS.keys():
```

Comment:
```suggestion
    if logging_level in LOGGING_LEVELS:
```

Reply:
Awesome, Thank you for this suggestion. I didn't know about this implementation for checking of the condition 😄